### PR TITLE
Modify AB3 Logic (Parser) 

### LINKED ISSUE #63: Discuss Status class implementation
## Details
Currently our implementation of our status class is an enum class. Regarding this, what are your thoughts on this and what implementation should we use? 
- an enum class or
- a normal class with enum implementation or
- a pure normal class.

### PR DESCRIPTION
**Pull request by ghost**
Modify the original address book parser and logic classes to fit Trackermon
List of modify (create a separate class with postfix Trackermon) classes:
1. AddCommandParser
2. DeleteCommandParser
3. Parser
4. ParserUtil
5. LogicManager
6. Logic
7. Status (change from enum to actual class for easiler ref for Parser classes)

### CODE REVIEW

File: src/main/java/seedu/address/logic/LogicTrackermon.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     *
+     * @see seedu.address.model.ModelTrackermon#getShowList()
+     */
+    ReadOnlyShowList getShowList();
+
+    /** Returns an unmodifiable view of the filtered list of shows */
+    ObservableList<Show> getFilteredShowList();
+
+    /**
+     * Returns the user prefs' address book file path.
```
Comment (arcornior) [inline src/main/java/seedu/address/logic/LogicTrackermon.java:37]: Should this be "Returns the user prefs' show list file path" instead?

File: src/main/java/seedu/address/model/show/Status.java [hunk 2]
```diff
@@ -21,9 +23,15 @@ public String toString() {
...
+
+    /**
+     * Returns true if a given string is a valid status.
+     */
+    public static boolean isValidStatus(String test) {
+        String stest = test.toLowerCase().trim();
+        return (stest.equals(COMPLETED) || stest.equals(WATCHING));
+    }
+
 }
```
Comment (arcornior) [inline src/main/java/seedu/address/model/show/Status.java:37]: Regarding Status, shall we first have a discussion on the change of implementation first?

**Review by arcornior — CHANGES_REQUESTED** [review]
Do remember to change your JavaDocs for LogicTrackermon, other than that, let's hold off on merging until discussion on Status class's implementation is finalised.

File: src/main/java/seedu/address/logic/LogicTrackermon.java (hunk 1 above)
Comment (lowkaiwei98) [reply src/main/java/seedu/address/logic/LogicTrackermon.java]: Updated

File: src/main/java/seedu/address/model/show/Status.java (hunk 2 above)
Comment (lowkaiwei98) [reply src/main/java/seedu/address/model/show/Status.java:37]: What do you suggest?

Comment (arcornior) [reply src/main/java/seedu/address/model/show/Status.java:37]: We've started a discussion thread under "Issues". Let's discuss there instead!

**Review by Ardentsoul — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/model/show/Status.java (hunk 2 above)
Comment (lowkaiwei98) [reply src/main/java/seedu/address/model/show/Status.java]: Updated

**Review by arcornior — APPROVED** [review]

**Pull request merged**